contracts/zaps/ZapInAllowance.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IporFusionZapIn {
+    /// @notice Returns the address of the current user performing a zap-in operation
+    function currentZapSender() external view returns (address);
+}
+
+/// @title ZapInAllowance
+/// @notice Helper contract for handling token allowances in IporFusionZapIn
```

Comment:
Fix all comments - not use Ipor Fusion name in ZapIn concept, because this contracts will be public good